cdap-docs/developers-manual/source/security/authorization-extensions.rst

```diff
@@ -8,4 +8,42 @@
 Authorization Extensions
 ========================
 
-Authorization backends in CDAP are implemented as extensions.
+Authorization backends in CDAP are implemented as extensions. Authorization extensions are loaded and executed in their
+own Java classloader. The reason for implementing them as extensions is so that a variety of existing authorization
+systems can be plugged-in to CDAP, and can execute using their own classloader without having to worry about conflicts
+with CDAP's system classloader. Each authorization extension is designed to be a fully self-contained JAR file,
+packaged with the required versions of all its dependencies. This is similar to how CDAP applications
+are packaged. In fact, users can use the CDAP application archetype to write their own authorization extension for
+CDAP. Having the required dependencies packaged within the JAR file ensures that there will be no conflicts even if
+CDAP itself uses a different version of the same library that the extension requires.
+
+Writing your own authorization extension
+----------------------------------------
+CDAP provides an authorization SPI (Service Provider Interface) for users to implement their own authorization
+extensions. To implement an authorization extension:
+
+- Create a maven project using the CDAP Application Archetype
+- Implement your authorization extension by extending the ``AbstractAuthorizer`` class
+- This class must be specified as the ``Main-Class`` attribute in the extension jar's manifest file. This can be done
+  automatically by specifying the ``app.main.class`` property to the fully qualified class name in the ``pom.xml`` of
```

Comment:
Oh we expect this to be set in the same way. Ignore my comment above about app.main.class